infra/app/api.bicep

```diff
@@ -0,0 +1,315 @@
+param name string
+param location string = resourceGroup().location
+param tags object = {}
+param aiService string = 'AzureOpenAI'
+param completionModel string = 'gpt-35-turbo'
+param embeddingModel string = 'text-embedding-ada-002'
+param plannerModel string = 'gpt-35-turbo'
+param memoryStore string
+param appServicePlanId string
+param appInsightsConnectionString string
+param azureCognitiveSearchName string
+param deployWebSearcherPlugin bool
+param allowedOrigins array = []
+param functionAppWebSearcherPluginName string
+param searcherPluginDefaultHostName string
+param openAIServiceName string
+param openAIEndpoint string
+param storageAccountName string
+param virtualNetworkId0 string
+param appServiceQdrantDefaultHost string
+param deployCosmosDB bool
+param deploySpeechServices bool
+param speechAccountName string
+param webApiClientId string
+param frontendClientId string
+param azureAdTenantId string
+param azureAdInstance string = environment().authentication.loginEndpoint
+param cosmosAccountName string
+param cosmosAccountEndpoint string
+
+var functionAppWebSearcherPluginId = resourceId(subscription().subscriptionId, resourceGroup().name,
+  'Microsoft.Web/sites', functionAppWebSearcherPluginName)
+var openAIId = resourceId(subscription().subscriptionId, resourceGroup().name,
+  'Microsoft.CognitiveServices/accounts', openAIServiceName)
+var storageAccountId = resourceId(subscription().subscriptionId, resourceGroup().name,
+  'Microsoft.Storage/storageAccounts', storageAccountName)
+var speechAccountId = resourceId(subscription().subscriptionId, resourceGroup().name,
+  'Microsoft.CognitiveServices/accounts', speechAccountName)
+var cosmosAccountId = resourceId(subscription().subscriptionId, resourceGroup().name,
+  'Microsoft.DocumentDB/databaseAccounts', cosmosAccountName)
```

Comment:
What we typically do I pass the name and then get a reference to the storage account with 'existing' keyword in bicep and then use that, 

like

resource storage '' = existing {
   name:
}

then 

storage.listKeys

Reply:
Can you see that is possible?

Reply:
It's possible, but based on your [comment](https://github.com/jongio/chat-with-your-data-solution-accelerator/pull/2#discussion_r1383732678) in another sample, the "existing" call is a costly ARM deployment, and there are five services that need to get keys by referencing "existing". Are you sure change these five services to reference the "existing" keyword?

Reply:
My point on that was that if we already have the reference then we shouldn't need to query it again.  Let's keep it as-is for now and not switch to existing.  Thanks